pkg/network/route_cache.go

```diff
@@ -158,7 +158,7 @@ func (n *netlinkRouter) Route(source, dest util.Address, netns uint32) (Route, b
 	r := routes[0]
 	log.Tracef("route for src=%s dst=%s: scope=%s gw=%+v if=%d", source, dest, r.Scope, r.Gw, r.LinkIndex)
 	return Route{
-		Gateway: util.AddressFromNetIP(r.Gw),
```

Comment:
Before this change, this would've resulted in an empty IPv6 `util.Address`, correct? AFAICT `IsUnspecified()` would've returned `true` for that, and the lookup would be skipped. I don't see how the behavior is actually changed. Can you clarify for me?

Reply:
It would have resulted in a `nil` `util.Address`, yes. The problem is `net.IP.IsUnspecified()` actually will not check for a `nil` `net.IP`, and will return `false` for it.

Reply:
The return value from `AddressFromNetIP` cannot be `nil`. It will be either `[4]byte` or `[16]byte`. Looking at https://github.com/DataDog/datadog-agent/blob/9ca9e76cc32fa1f457da788e5dcf3bb9748b752d/pkg/process/util/address.go#L17-L27 it looks like it would return a `v6address` aka `[16]byte`, which would be all zeroes.

Reply:
Ok, yes, it would be all zeros. Something else is up with this code then because we keep spamming the ec2 metadata endpoint.